### PR TITLE
PLT-6842 Autodeploy on push

### PR DESCRIPTION
**Pull request by tshaynik**
https://input-output.atlassian.net/browse/PLT-8402

Autodeploy to Kubernetes on push.
This PR adds a Github workflow similar to what currently exists in https://github.com/input-output-hk/marlowe-token-plans and https://github.com/input-output-hk/marlowe-playground/ for updating the values.yaml of the helm chart with the correct image tag and auto-committing, causing a deployment in KubeVela.

The `qa` environments track the latest commit. The `demo` environments are updated when a new version is tagged.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] [Test report is updated](https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe/test/test-report.md) (if relevant)
    - [ ] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, PNG optimization, etc. are updated
    - [ ] Operables are updated with changes to executable command line options.
    - [ ] Deploy charts updated with changes to operables.
- PR
    - [x] Self-reviewed the diff
    - [x] Useful pull request description
        - Review required
        - [ ] Substantial changes to code, test, or documentation
        - [ ] Change made to Marlowe validator (@bwbush and @palas must be included as reviewers)
        - Review not required
        - [ ] Minor changes to non-critical code, documentation, nix derivations, configuration files, or scripts
        - [ ] Formatting, spelling, grammar, or reorganization
    - [x] Reviewer requested

### CODE REVIEW

**Comment by tshaynik** [comment]
This is ready to review. It's still a draft just as a reminder to make a couple of changes before merging: only run on the `main` branch and to commit to `main`.

**Comment by shlevy** [comment]
@tshaynik In other repos with this approach to CD, the image deployment is part of the vela deployment pipeline. Here it's not, do we have some way to wait for the GHA image deployment to finish before kicking off the vela deploy?

**Review by shlevy — APPROVED** [review]

**Comment by tshaynik** [comment]
This should be ready to merge now.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -90,10 +95,43 @@ jobs:
...
+          ref: main
+
+      - name: Update publish version
+        id: update
+        run: |
+          if [[ "${{ github.ref }}" == "refs/tags/runtime@v"* ]]
+          then
+            export TAG=$(echo "${{ github.ref_name }}" | sed -e "s/^runtime@v//")
+            echo TAG=$TAG
+            yq e -i '.instances.demo.webTag = env(TAG)' deploy/marlowe-runtime/values.yaml
```
Comment (jhbertra) [inline .github/workflows/deploy.yml:117]:
```suggestion
            yq e -i '.instances.demo.tag = env(TAG)' deploy/marlowe-runtime/values.yaml
            yq e -i '.instances.demo.webTag = env(TAG)' deploy/marlowe-runtime/values.yaml
```

**Review by jhbertra — CHANGES_REQUESTED** [review]
Looks good, just needs one change

**Pull request merged**